python/lsst/analysis/drp/makeQaTractTables.py

```diff
@@ -0,0 +1,284 @@
+    these new tables are sorted in the same order as in the objectTable_tract
+    tables and, since we don't want to make assumptions on that ordering, we
+    load in the table itself to do the sorting.
+
+    By default, it is assumed that it is the values from the forced catalogs
+    (i.e. "forced_src") that are of interest and, in this case, certain columns
+    from the "meas" and "ref" catalogs that don't get propagated to the
+    "forced_src" catalogs are added to the persisted table, which has dataset
+    type "qaTractTable_forced".  Alternatively, the values from just the
+    unforced, "meas", catalogs can be requested by passing the config parameter
+    -c  makeQaTractTables:dfDataset="meas".  In this case, the persisted table
+    has dataset type "qaTractTable_unforced".
+    """
+
+    ConfigClass = MakeQaTractTablesTaskConfig
+    _DefaultName = "makeQaTractTablesTask"
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        measTypeStr = "forced" if self.config.dfDataset == "forced_src" else "unforced"
+        inputs["band"] = butlerQC.quantum.outputs["qaTractTable_" + measTypeStr][0].dataId["band"]
+        dataId = inputRefs.inputObjCats[0].datasetRef.dataId
+        inputs["tract"] = dataId["tract"]
+        outputs = self.run(**inputs)
+        butlerQC.put(outputs, outputRefs)
+
+    def run(self, inputObjCats, inputObjectTable, tract, band):
+        """Collect and aggregate catalog columns required for QA purposes
+
+        Parameters
+        ----------
+        inputObjCats : `list` of `pandas.core.frame.DataFrame`
+            The input _obj tables from which to collect the relevant data
+            columns.
+        tract : `int`
+            The tract for which to aggregate the tables in the requested
+            patch list.
+        band : `str`
+            The band for which to aggregate the input tables.
+
+        Raises
+        ------
+        RuntimeError
+            Raised if config.filterMap was provided, but there is no entry for
+            the band requested.
+        RuntimeError
+            Raised if an unknown dataset type was requested via
+            config.dfDataset.   Must be one of \"forced_src\" or \"meas\".
+        RuntimeError
+            Rased if no catalogs were read in for the given tract.
+
+        Returns
+        -------
+        allCats : `lsst.pipe.base.Struct`
+            Contains the dataFrame with the collated QA table for ``tract`` and
+            ``band``.
+        """
+        if self.config.filterMap is not None:
+            if band in self.config.filterMap:
+                filterName = self.config.filterMap[band]
+            else:
+                raise RuntimeError("A config.filterMap was provided, but there is no entry for the "
+                                   "band requested: {}".format(band))
+        else:
+            filterName = band
```

Comment:
Now that I see how you're using `filterMap`, I don't actually understand why you'd want to: you're starting with a `band` value that's what is directly used to identify the coadd (note that in Gen3, a coadd data ID uses `band`, _not_ `physical_filter`, because multiple `physical_filter` values can and often do go into one coadd).

So, by switching to `physical_filter`-like names, you're actually making the plots lie sometimes: they're saying that a coadd that includes one or more HSC-R2 visits is an HSC-R coadd, and that's not true.  I'm sure this dates back to before we started trying to be consistent about what labels we use for different kinds of filters (the Gen2 coadds are definitely not consistent), but the good news is that the consistent thing is also easier here.

Anyhow, I think you just want to use `band` values in all of the plots and column names (at least for anything derived from coadd measurements), and drop the `filterMap` config option entirely.